core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -885,9 +885,15 @@ sealed abstract class Chunk[+A] extends ChunkLike[A] with Serializable { self =>
   override def toArray[A1 >: A: ClassTag]: Array[A1] = {
     val dest = Array.ofDim[A1](self.length)
 
-    self.toArray(0, dest)
-
-    dest
+    try {
+      self.toArray(0, dest)
+      dest
+    } catch {
+      case _: ClassCastException =>
```

Comment:
Added this as fallback _just in case_ there are issues with other combinators

Reply:
Seems unlikely 🙂